src/soundbase.cpp

```diff
@@ -94,114 +102,9 @@ QString CSoundBase::SetDev ( const QString strDevName )
-    QString strReturn = "";
-
-    // init flag for "load any driver"
-    bool bTryLoadAnyDriver = false;
-
-    // check if an ASIO driver was already initialized
-    if ( !strCurDevName.isEmpty() )
-    {
-        // a device was already been initialized and is used, first clean up
-        // driver
-        UnloadCurrentDriver();
-
-        const QString strErrorMessage = LoadAndInitializeDriver ( strDevName, false );
-
-        if ( !strErrorMessage.isEmpty() )
-        {
-            if ( strDevName != strCurDevName )
-            {
-                // loading and initializing the new driver failed, go back to
-                // original driver and create error message
-                LoadAndInitializeDriver ( strCurDevName, false );
-
-                // store error return message
-                strReturn = QString ( tr ( "The selected audio device could not be used "
-                    "because of the following error: " ) ) + strErrorMessage +
-                    QString ( tr ( " The previous driver will be selected." ) );
-            }
-            else
-            {
-                // loading and initializing the current driver failed, try to find
-                // at least one usable driver
-                bTryLoadAnyDriver = true;
-            }
-        }
-    }
-    else
-    {
-        if ( !strDevName.isEmpty() )
-        {
-            // This is the first time a driver is to be initialized, we first
-            // try to load the selected driver, if this fails, we try to load
-            // the first available driver in the system. If this fails, too, we
-            // throw an error that no driver is available -> it does not make
-            // sense to start the software if no audio hardware is available.
-            if ( !LoadAndInitializeDriver ( strDevName, false ).isEmpty() )
-            {
-                // loading and initializing the new driver failed, try to find
-                // at least one usable driver
-                bTryLoadAnyDriver = true;
-            }
-        }
-        else
-        {
-            // try to find one usable driver (select the first valid driver)
-            bTryLoadAnyDriver = true;
-        }
-    }
-
-    if ( bTryLoadAnyDriver )
-    {
-        // if a driver was previously selected, show a warning message
-        if ( !strDevName.isEmpty() )
-        {
-            strReturn = tr ( "The previously selected audio device "
-                "is no longer available or the audio driver properties have changed to a state which "
-                "is incompatible with this software. We now try to find a valid audio device. This new "
-                "audio device might cause audio feedback. So, before connecting to a server, please "
-                "check the audio device setting." );
-        }
-
-        // try to load and initialize any valid driver
-        QVector<QString> vsErrorList = LoadAndInitializeFirstValidDriver();
```

Comment:
Can we ask the user if he wants to do this?

Reply:
I've added a button in the settings window to do this.